src/lib/ngsi/Metadata.cpp

```diff
@@ -259,11 +259,14 @@ std::string Metadata::render(const std::string& indent, bool comma)
 
     if (compoundValueP->isObject())
     {
+      // Note in this case we don't add the "value" key, the toJson()
+      // method does it for toplevel compound (a bit crazy... this deserves a FIXME mark)
+      compoundValueP->renderName = true;
       part = compoundValueP->toJson(true, false);
     }
     else if (compoundValueP->isVector())
     {
-      part = "[" + compoundValueP->toJson(true, false) + "]";
+      part = JSON_STR("value") + ": [" + compoundValueP->toJson(true, false) + "]";
```

Comment:
Oh yes, we have plenty of tests.
The "value": is added before this function.
There is a risk we'll get a double "value": with this change.
I know this because I've been here seeing/fixing problems not so long ago.
However, I imagine all functests work so we should be safe.

In the case of an JSON Object, the rendering is different, no "value": there ...
We should go over this and clean it, amke Array and Object behave the same way.

I just wonder, is it ok to hardcode "value" here?
What if we have a name of the vector? For keyValues ...
In any case, if all functests are OK, all is OK. At least until we go deeper here

Perhaps we should add an issue about 'refactoring' this, unless we already have one (I might have created one not too long ago)

NTC